fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/impl/resource/conditions/OverlayConditionsMetadata.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.resource.conditions;
```

Comment:
Why isn't this a public API class? Without this class pack.mcmeta datagen is impossible

Reply:
Oops, I see this was asked to be changed a month ago — in my defence, the PR kept getting closed and it was a bit hard to keep track of.

In any case I still think this class should be an API anyway

Reply:
We should definally be able to datagenerate this, however I worry just moving this to API isnt the ideal solution.
- Generally we dont make records/data classes part of the API, we would have an interface + static factory function, it provides a better API surface.
- How easy is it to datagenerate the pack.mcmeta file right now? I dont think we do anything special for it, this does sound like a useful thing to have. Maybe its best leaving this as-is for now and making a nice API for generating the file?

Reply:
> Generally we dont make records/data classes part of the API
However, this class is very similar to (even copies) the vanilla `PackOverlaysMetadata`, which makes it easy to understand how to use it

> How easy is it to datagenerate the pack.mcmeta file right now?
The only thing is `ResourceFilter` requires AW for `BlockEntry::<init>`, but that's the only such case. I datagen `pack.mcmeta` to filter recipes & advancemenets

Reply:
Well, actually, the static factory function sounds ok, the main thing is to give a name similar to the vanilla class (yarn)

Reply:
Going to merge this PR as-is, we can have a follow up PR making this better for datagen.